### PR TITLE
add oled and encoder function to lck75

### PR DESCRIPTION
**Pull request by lyso1**
## Description

separated the OLED and encoder code into their own files and added `#include oled.c` & `#inlcude encoder.c` into the lck75.c file so that the configurator includes the desired code when compiling.
 having the code in the keymap file didn't work and I found that the configurator will use any code so long it is in the keyboard folder-based and not keymap-based.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

*  enables configurator to compile the set encoder & oled code for the lck75.c

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by tzarc** [comment]
Having the code in the keymap allows for other users who don't use configurator to customise the build.
With your changes you're forcing everyone to use the OLED in the way specified in the default keymap.

I don't think this is the way this should be handled.

File: keyboards/lck75/lck75.c [hunk 1]
```diff
@@ -14,4 +14,140 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "lck75.h"
\ No newline at end of file
+
+#include "lck75.h"
+
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/lck75/lck75.c:20]:
```suggestion
__attribute__((weak)) void encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/lck75/lck75.c [hunk 2]
```diff
@@ -14,4 +14,140 @@
...
+        } else {
+            if (timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+                anim_timer = timer_read32();
+                animation_phase();
+            }
+        }
+    }
+}
+#ifdef OLED_DRIVER_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```
Comment (drashna) [inline keyboards/lck75/lck75.c:146]:
```suggestion
__attribute__((weak)) oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
> Having the code in the keymap allows for other users who don't use configurator to customise the build.
> With your changes you're forcing everyone to use the OLED in the way specified in the default keymap.
> 
> I don't think this is the way this should be handled.

As long as the functions are defined weakly, I think that should be okay.

File: keyboards/lck75/lck75.c [hunk 3]
```diff
@@ -14,4 +14,140 @@
...
+#define ANIM_FRAME_DURATION 200
+#define ANIM_SIZE 512
+
+bool gui_on = true;
+uint32_t anim_timer = 0;
+uint32_t anim_sleep = 0;
+uint8_t current_idle_frame = 0;
+uint8_t current_tap_frame = 0;
+
+static void render_anim(void) {
```
Comment (drashna) [inline keyboards/lck75/lck75.c:45]: I would say move all of this into the oled_task_user function.

**Comment by lyso1** [comment]
currently having the OLED code modified

File: keyboards/lck75/lck75.c (hunk 3 above)
Comment (lyso1) [reply keyboards/lck75/lck75.c]: updated the oled code

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**